eclipse-jdt/eclipse.jdt.ui
# Move LambdaExpressionAndMethodRefCleanUp into jdt.core.manipulation

## What it does

* Moves the "LambdaExpressionAndMethodRef" clean up logic to `org.eclipse.jdt.core.manipulation` so that it can be used in jdt.ls as well.
* Adds method in `LambdaExpressionAndMethodRefFixCore` to provide the clean up fix for a given `ASTNode`


## How to test

* Confirm that `LambdaExpressionAndMethodRefFixCore` can be accessed without needing to import any jdt.ui packages.
* Confirm that `CleanUpTest` passes


## Author checklist
